### ISSUE TITLE
[feature]: how to use sha256 in Pre-request Scripts Scripts

### ISSUE THREAD

**Issue by zlay0701** [hoppscotch/hoppscotch#2015]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

how to use sha256 in Pre-request Scripts Scripts?

### Why should this be worked on?

Some interfaces need to dynamically generate signatures and initiate requests according to tokens
The signature algorithm uses algorithms such as sha256 and SHA1

**Comment by liyasthomas** [comment]
Similar issue: #2001
Adding support to npm packages and SDKs are in our roadmap. Stay tuned for product updates.

**Comment by apptut** [comment]
npm packages +1, I want gen sign with params by md5 or sha1

**Comment by AndrewBastin** [comment]
Hate to burst the bubble, but I am not planning on getting NPM packages as is to work with Hoppscotch scripting environment.
It's a complicated enough task as is and implementing mechanisms to manage NPM and node_modules and dependency management and Node Module Resolution capabilities (remember, the scripts are executed client side in a sandbox) all within the sandbox is a massive challenge which we cannot spend too much effort on.

But, I was hoping to get Deno style ESM URL imports working so you can directly import from URL (read more here: [https://deno.land/manual/examples/import_export](https://deno.land/manual/examples/import_export)) and there are sources like [JSPM](https://jspm.io), [Skypack](https://skypack.dev) and [Unpkg (as an option)](https://unpkg.com) which give all the important libraries in the ESM format.

**Comment by apptut** [comment]
I agree with what you said. My purpose is not to import npm, but to get the capabilities of encryption libraries. Many times the interface needs to be signed using md5, or sha1, base64 algorithms, and so on.

**Comment by zlay0701** [comment]
can add https://github.com/brix/crypto-js ?

**Comment by evanlurvey** [comment]
I also have needed crypto-js before for auth and signatures.

**Comment by sneha-herle** [comment]
Please find a solution for this, needed crypto-js before for auth and signatures.

**Comment by bappleug-tw** [comment]
Plus one for adding crypto-js support

**Comment by jhoos70** [comment]
Need crypto-js, too. Without it hoppscotch is useless for my usecase.

**Comment by samuel-deal-tisseo** [comment]
Plus one for adding crypto-js support

**Comment by qichengri** [comment]
Plus one for adding crypto-js support

**Comment by archcorsair** [comment]
Any updates on this? This feature is the only one preventing us from moving from postman to hoppscotch

**Comment by siniradam** [comment]
Seems like some crypto features available in the browser API working, if you someone is in dire need of HMAC, maybe can give this this a try.

```js
async function createHMAC(payload, secret) {
  if (typeof payload == "object") {
    payload = JSON.stringify(payload);
  }
  let enc = new TextEncoder("utf-8");
  let algorithm = { name: "HMAC", hash: "SHA-256" };
  let key = await crypto.subtle.importKey("raw", enc.encode(secret), algorithm, false, [
    "sign",
    "verify",
  ]);
  let signature = await crypto.subtle.sign(algorithm.name, key, enc.encode(payload));
  let digest = btoa(String.fromCharCode(...new Uint8Array(signature)));
  return digest;
}
```

**Comment by alizufan** [comment]
> Seems like some crypto features available in the browser API working, if you someone is in dire need of HMAC, maybe can give it a try to this function.
> 
> ```js
> async function createHMAC(payload, secret) {
>   if (typeof payload == "object") {
>     payload = JSON.stringify(payload);
>   }
>   let enc = new TextEncoder("utf-8");
>   let algorithm = { name: "HMAC", hash: "SHA-256" };
>   let key = await crypto.subtle.importKey("raw", enc.encode(secret), algorithm, false, [
>     "sign",
>     "verify",
>   ]);
>   let signature = await crypto.subtle.sign(algorithm.name, key, enc.encode(payload));
>   let digest = btoa(String.fromCharCode(...new Uint8Array(signature)));
>   return digest;
> }
> ```

thank you !

**Comment by jamesgeorge007** [comment]
Hi! #5090 added the ability to import ESM packages in the scripting context as part of the `Experimental scripting sandbox`. You can now import crypto libraries directly:

```ts
import CryptoJS from 'https://esm.run/crypto-js@4.2.0';
```

We're aware of certain limitations with the native `crypto` module. ESM imports offer an alternative approach. Ongoing scripting improvements are tracked in #5221.

In the meantime, you can choose to proceed with either of the following:
- Use ESM imports with the `Experimental scripting sandbox`.
- Opt out of the experimental scripting sandbox from settings to use the native `crypto` module.
  <img width="1906" height="565" alt="Image" src="https://github.com/user-attachments/assets/bb540bb1-4ec4-46b1-99be-da832a7c6adc" />

**Comment by jamesgeorge007** [comment]
Closing as this is addressed in [v2026.1.0](https://github.com/hoppscotch/hoppscotch/releases/tag/2026.1.0) via #5791.
